src/topology/vector_bundle.lean

```diff
@@ -391,3 +406,100 @@ lemma is_open_map_proj : is_open_map Z.proj :=
 topological_fiber_bundle_core.is_open_map_proj Z
 
 end topological_vector_bundle_core
+
+end
+
+section
+
+/-! ### Topological vector prebundle -/
+
+variable [∀ x, topological_space (E x)]
+
+open topological_space
+
+/-- This structure permits to define a vector bundle when trivializations are given as local
+equivalences but there is not yet a topology on the total space. The total space is hence given a
+topology in such a way that there is a fiber bundle structure for which the local equivalences
+are also local homeomorphism and hence vector bundle trivializations. -/
+@[nolint has_inhabited_instance]
+structure topological_vector_prebundle :=
+(pretrivialization_at : B → topological_vector_bundle.pretrivialization R F E)
+(mem_base_pretrivialization_at : ∀ x : B, x ∈ (pretrivialization_at x).base_set)
+(continuous_triv_change : ∀ x y : B, continuous_on ((pretrivialization_at x) ∘
+  (pretrivialization_at y).to_local_equiv.symm) ((pretrivialization_at y).target ∩
+  ((pretrivialization_at y).to_local_equiv.symm ⁻¹' (pretrivialization_at x).source)))
+(total_space_mk_inducing : ∀ (b : B), inducing ((pretrivialization_at b) ∘ (total_space_mk E b)))
+
+namespace topological_vector_prebundle
+
+variables {R E F}
+
+/-- Natural identification of `topological_vector_prebundle` as a `topological_fiber_prebundle`. -/
+def to_topological_fiber_prebundle (a : topological_vector_prebundle R F E) :
+  topological_fiber_prebundle F (proj E) :=
+{ pretrivialization_at := λ x, a.pretrivialization_at x, ..a }
+
+/-- Topology on the total space that will make the prebundle into a bundle. -/
+def total_space_topology (a : topological_vector_prebundle R F E) :
+  topological_space (total_space E) :=
+a.to_topological_fiber_prebundle.total_space_topology
+
+/-- Promotion from a `topologial_vector_prebundle.trivialization` to a
+  `topologial_vector_bundle.trivialization`. -/
+def trivialization_at (a : topological_vector_prebundle R F E) (x : B) :
+  @topological_vector_bundle.trivialization R _ F E _ _ _ _ _ _ _ a.total_space_topology  :=
+begin
+  letI := a.total_space_topology,
+  exact { linear := (a.pretrivialization_at x).linear,
+  ..a.to_topological_fiber_prebundle.trivialization_at x }
+end
+
+variable (a : topological_vector_prebundle R F E)
+
+lemma mem_trivialization_at_source (b : B) (x : E b) :
+  total_space_mk E b x ∈ (a.pretrivialization_at b).source :=
+begin
+  simp only [(a.pretrivialization_at b).source_eq, mem_preimage, proj],
+  exact a.mem_base_pretrivialization_at b,
+end
+
+lemma total_space_mk_preimage_soure (b : B) :
+  (total_space_mk E b) ⁻¹' (a.pretrivialization_at b).source = univ :=
```

Comment:
Possibly an easier way to work with this would be:
```suggestion
  ∀ x, total_space_mk E b x ∈ (a.pretrivialization_at b).source :=
```

Reply:
I actually really want the statement above, now I changed it to a simp lemma. I want the above expression to be simplified to `univ`